File: solutions/TypeScript/[1446_Easy] Consecutive Characters/maxPower.ts
function maxPower(str: string): number {
  let previousCharacter = '';
  let currentCount = 0;
  let maxCount = 0;

  for (const character of str) {
    if (previousCharacter !== character) {
      if (currentCount > maxCount) {
        maxCount = currentCount;
      }
      currentCount = 0;
      previousCharacter = character;
    }

    currentCount++;
  }

  return currentCount > maxCount ? currentCount : maxCount;
}

File: solutions/Java/[1446_Easy] Consecutive Characters/maxPower.java
class Solution {
  public int maxPower(String s) {
    char[] characters = s.toCharArray();
    char previousChar = characters[0];
    int currentCount = 0;
    int maxCount = 0;

    for (char c: characters) {
      if (previousChar != c) {
        if (currentCount > maxCount) {
          maxCount = currentCount;
        }
        currentCount = 0;
        previousChar = c;
      }

      currentCount++;
    }

    return currentCount > maxCount ? currentCount : maxCount;
  }
}
